test/jdk/java/lang/invoke/defineHiddenClass/UnloadingTest.java

```diff
@@ -215,17 +215,13 @@ private HiddenClassUnloader(Class<?> hc) {
 
         void unload() {
             // Force garbage collection to trigger unloading of class loader and native library
-            ForceGC gc = new ForceGC();
-            assertTrue(gc.await(() -> weakRef.get() == null));
-
-            if (weakRef.get() != null) {
+            if (!ForceGC.wait(() -> weakRef.get() == null)) {
                 throw new RuntimeException("loader " + " not unloaded!");
             }
         }
 
         boolean tryUnload() {
-            ForceGC gc = new ForceGC();
-            return gc.await(() -> weakRef.get() == null);
+            return ForceGC.wait(() -> weakRef.get() == null, 1000L);
```

Comment:
This verifies that the reference will never become weakly unreachable, i.e. it remains strongly reachable.

I suggest to add a method default to 200ms (?) timeout scaled with the timeout factor and expect the specified condition will never met.   Maybe call it `ForceGC.neverTrue(BooleanSupplier)`?   Need to tune the base waiting time; for example if timeout factor is 4, then total wait would be 800 ms.